driver/src/main/java/org/neo4j/driver/internal/util/AddressUtil.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.neo4j.driver.internal.util;
+
+import java.util.regex.Pattern;
+
+public class AddressUtil
+{
+    private static final Pattern LOCALHOST = Pattern.compile( "^(localhost|127(\\.\\d+){3})$", Pattern.CASE_INSENSITIVE );
+
+    /**
+     * Return true if the host provided matches "localhost" or "127.x.x.x".
+     *
+     * @param host the host name to test
+     * @return true if localhost, false otherwise
+     */
+    public static boolean isLocalhost( String host )
+    {
+        return LOCALHOST.matcher( host ).matches();
```

Comment:
Multiple network cards don't apply here. We're talking about the address to which the client connects and whether that is equal to `localhost` or `127.x.x.x`.

I'm also not sure that `127.99999.99.123123123123` would actually be a valid IP address anyway. But I'll check out the utility methods you mention. We do need to make sure we catch `localhost` here as well as the raw IP addresses so I'll work out which one(s) do that.